src/enclave/mod.rs

```diff
@@ -15,3 +15,100 @@ mod ioctls;
 
 pub use builder::*;
 pub use enclave::*;
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    use crate::types::page;
+    use crate::types::tcs::Tcs;
+
+    use std::num::NonZeroU32;
+
+    use lset::Span;
+    use primordial::Page;
+
+    #[allow(clippy::erasing_op)]
+    const CODE_OFFSET: usize = Page::size() * 0;
+    const TCS_OFFSET: usize = Page::size() * 1;
+    const SSA_OFFSET: usize = Page::size() * 2;
+    const SSA_COUNT: NonZeroU32 = unsafe { NonZeroU32::new_unchecked(1) };
+
+    /// Don't call this function. It has a non-standard ABI.
+    #[naked]
+    unsafe extern "C" fn code() -> ! {
+        asm!(
+            "0:",
```

Comment:
Oh, btw, you should not use `0` or `1`. See https://doc.rust-lang.org/nightly/unstable-book/library-features/asm.html#labels

> Moreover, due to an llvm bug, you shouldn't use labels exclusively made of 0 and 1 digits, e.g. 0, 11 or 101010, as they may end up being interpreted as binary values.